### ISSUE TITLE
GPU's are not detected

### ISSUE THREAD

**Issue by Tugcga** [Tugcga/Sycles#31]
In Sycles 2.2.0 only CPU is available as render device. In some machines GPU's are detected and can be used, but in some systems they are not visible. Some investigations, why it happens, are required.

**Comment by StefanLaschet** [comment]
Are you still working on the GPU problem? I just installed the new sycles version 2.2.0 and there is no gpu device to select. And the cpu rendering ist not really an alternative. 
Thanks a lot for your help.
Best regards

**Comment by Tugcga** [comment]
I still have no idea how to fix it.

**Comment by StefanLaschet** [comment]
Ok, I hope you will find it out. 

And if I use the camera sequencer Editor in Softimage with Cycles 2.1.3, there is no depth of field in the Renderings. Is this solved in the new version?
Thanks a lot and best regards,
Stefan

**Comment by StefanLaschet** [comment]
I just tried the new Version on my Office-PC and there I can chose the graphic card as a device. But the rendering takes much longer than with the previous version. It seems, that it still renders only with the cpu. Maybe this helps you finding the error. 
Thanks a lot and best regards, Stefan

**Comment by RuiSantos1970** [comment]
Hi there,
Just to let you know i have the same problem, Cycles is not detecting the GPU,only CPU is showing
BTW,Thank you for your work on this!
Rui

**Comment by StefanLaschet** [comment]
Hi,
thanks a lot for the new Sycles Version. I am so greatful that you take your time and talent to program this renderer for Softimage. In my opinion it is still the best 3D Software.
And maybe you can help me with another problem. When I open a render region, Softimage freezes very often. It doesn't depent if I use mental ray or Sycles. Sometimes it works quite well but very often, when the region is open and I change some values, the software freezes. Do you have an idea?
And is there still the gpu problem in the new version?
Thanks a lot and best regards, Stefan

**Comment by Tugcga** [comment]
You can try to disconnect all workgroups in the Softimage and uninstall all addons. The activate it one by one and check when freezes are appear. It looks like some addon executes some process before the render starts.

The status about gpu rendering in Sycles is the same.

**Comment by RuiSantos1970** [comment]
Hi Tugcga,
well i did what you've suggested.I backed up the softimage user preferences
files at /user/autodesk/softimage....created a clean,new one without any
addons..
installed cycles 2.2.1,and in the options menu only the cpu is visible.
I went back tried 2.2.0,the same result...the one that worked was the 2.1.3
version...so something changed from 2.1.3 to 2.2.0 that makes this bug
active.
Thank you for your work!
Rui



Rui Santos - Animação3D
www.ruisantos3d.com | Instagram <https://www.instagram.com/ruisantos3d/>  |
***@***.***
Telef. 00351 969612580 | 920552064
------------------------------------------------------------------------------------------------------
Rui Santos 360º
Visitas Virtuais 360º
www.ruisantos360.com | ***@***.*** |
Telefone: +351 920552064


Tugcga ***@***.***> escreveu (quinta, 3/10/2024 à(s) 11:12):

**Comment by wireframex** [comment]
Hello,
I tested several configurations without conclusive results:
- I replaced the libraries from Blender (several Alpha / beta sources ...)
- I replaced several DLLs including the Cuda DLLs
- ...
However, I noticed something that may help:
- The latest Blender version 4.2.3 LTS recognizes my GPUs perfectly but not the last BForartist version (https://www.bforartists.de/) there may be a lead here ?

For information, I tested the Sycles 2.2.1 version with:
- 1  computer with RTX 3090 cards under Windows 11
- 1 laptop with GPU 4090 under Windows 11 (Asus Zephyrus GX650PY)
Only version 2.1.3 works perfectly

Bests regards
Phil

**Comment by wireframex** [comment]
Hey Stefan,
It actually doesn't work at all for  SI :(
I came back to Sycles_2_1_3

**Comment by OurMoods** [comment]
@Tugcga  from the Cycles building doc, there is a "-DWITH_LEGACY_LIBRARIES=ON" toggle when building the binaries . 
Don't know much about it, but maybe it will provide some clue of the problem? :)

**Comment by Tugcga** [comment]
It looks like this parameter is required when building Cycles with Hydra support. It's not needed for the Softimage addon. In any case, I have not been able to build Cycles with the WITH_LEGACY_LIBRARIES parameter enabled. This may require some additional investigation.

**Comment by OurMoods** [comment]
> In any case, I have not been able to build Cycles with the WITH_LEGACY_LIBRARIES parameter enabled. 
@Tugcga 
Yes, the building doc says it is needed ONLY for Hydra and Houdini.
In other word, it means: **EVEN Houdini is KNOWN** to have issues with Cycles current dependents, let alone our old SI :)

plus, let you once said, current the denoise lib ALSO not loaded, so that you implement it using other method instead of build-in function.

So I just guess things break after 2.1.3 are due to the update of dependence lib, that is why I just point out this "LEGACY_LIBRARIES", and hope this is a correct clue/direction : ) 

![Image](https://github.com/user-attachments/assets/dbe552b1-1d85-4afd-aa77-d7fc0a2339f0)
